README.md

```diff
@@ -5,29 +5,78 @@
+    * [app container](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose.yml#L62-L81) sets up the [Hyrax] application, which is then used by 2 services - web and workers.<br/><br/>
+    * [Web container](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose.yml#L83-L94) runs the materials data repository application. By default, this runs the materials data repository service on port 3000 internally in docker (http://web:3000). <br/><br/>This container runs [docker-entrypoint.sh](https://github.com/antleaf/nims-hyrax/blob/develop/hyrax/docker-entrypoint.sh). It needs the database, solr and fedora containers to be up and running. It waits for 15s to ensure Solr and fedora are running and exits if they are not. It [runs a rake task](https://github.com/antleaf/nims-hyrax/blob/develop/hyrax/docker-entrypoint.sh#L38-L39), ([setup_hyrax.rake](https://github.com/antleaf/nims-hyrax/blob/develop/hyrax/lib/tasks/setup_hyrax.rake)) to setup the application. <br/><br/>The default workflows are loaded, the default admin set and collection types are created and the users in [setup.json](https://github.com/antleaf/nims-hyrax/blob/develop/hyrax/seed/setup.json) are created as a part of the setup.<br/><br/>
+    * [Wokers container](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose.yml#L96-L107) runs the background tasks for materials data repository, using [sidekiq](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=2ahUKEwio06ew2qPhAhUMZt4KHT0jDwQQFjAAegQIBBAB&url=https%3A%2F%2Fgithub.com%2Fmperham%2Fsidekiq&usg=AOvVaw3mZXHmVT7i5YYB8_u56eH2) and redis. By default, this runs the worker service. <br/><br/> Hyrax processes long-running or particularly slow work in background jobs to speed up the web request/response cycle. When a user submits a file through a work (using the web or an import task), there a humber of background jobs that are run, initilated by the hyrax actor stack, as explained [here](https://samvera.github.io/what-happens-deposit-2.0.html)<br/><br/>You can monitor the background workers using the materials data repository service at http://web:3000/sidekiq when logged in as an admin user.
+  * [docker-compose.override.yml](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose.override.yml) This file exposes the ports for fcrepo, solr and the hyrax web container, so they an be accessed outside the container. If running this servcie in development or test, we could use this file.
+  * [docker-compose-production.yml](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose-production.yml) builds the [nginx container](https://github.com/antleaf/nims-hyrax/blob/develop/docker-compose-production.yml#L15-L26) running the nginx service, which will reverse proxy requests from the web service, run by the web container. This will expose port 443 to the users, so they can interact with the materials data repository using https://mdr-domain-name.com
```

Comment:
it will also fetch a certbot / letsencrypt SSL certificate. Could be a bit tricky behind a firewall.